reladomo/src/main/java/com/gs/fw/common/mithra/transaction/TransactionOperation.java

```diff
@@ -255,6 +262,24 @@ protected boolean touchesSameObject(TransactionOperation otherOp)
         return loopCompare(this.getAllObjects(), otherOp.getAllObjects());
     }
 
+    private boolean touchesSameObjectUsingIndex(TransactionOperation biggerOperation, TransactionOperation smallerOperation) {
+        RelatedFinder finder = this.getPortal().getFinder();
+        if (finder.getAsOfAttributes() == null)
+        {
+            return indexCompare(biggerOperation.getIndexedObjects(), smallerOperation.getAllObjects());
+        }
+        //for temporal objects can't use the unique index with as-of attributes
+        Extractor[] extractors = finder.getPrimaryKeyAttributes();
+        List allObjectsFromBiggerOperation = biggerOperation.getAllObjects();
+        FullUniqueIndex index = new FullUniqueIndex(extractors, allObjectsFromBiggerOperation.size());
+        index.setUnderlyingObjectGetter(new TransactionalUnderlyingObjectGetter());
+        for (int i =0 ; i < allObjectsFromBiggerOperation.size(); i++)

```

Comment:
Why are we indexing the big operation? It would be more efficient to index the smaller one.

Reply:
That's a good point and I was wondering about it too. The size is determined by getTotalOperationsSize() instead of getAllObjects().size(). Does that have any bearing?

Reply:
The number of items in the index is the only thing that determines the efficiency. Maybe you should leave the big/small determination to the touchesSameObjectUsingIndex method itself.

Reply:
Makes sense. Let me change it.